### ISSUE TITLE
Breaking in Angular 6

### ISSUE THREAD

**Issue by grosch** [SheetJS/js-xlsx#1103]
I've just upgrade my project to Angular 6 and now i'm getting this error:

xlsx.js:11 Uncaught ReferenceError: global is not defined
    at make_xlsx (xlsx.js:11)
    at Object../node_modules/xlsx/xlsx.js (xlsx.js:19856)
    at __webpack_require__ (bootstrap:81)
    at Object../src/app/metric/network-remediation/network-remediation.component.ts (main.js:2187)
    at __webpack_require__ (bootstrap:81)
    at Object../src/app/app.module.ts (app.component.ts:8)
    at __webpack_require__ (bootstrap:81)
    at Object../src/main.ts (environment.ts:10)
    at __webpack_require__ (bootstrap:81)
    at Object.0 (main.ts:11)
make_xlsx @ xlsx.js:11
./node_modules/xlsx/xlsx.js @ xlsx.js:19856
__webpack_require__ @ bootstrap:81
./src/app/metric/network-remediation/network-remediation.component.ts @ main.js:2187
__webpack_require__ @ bootstrap:81
./src/app/app.module.ts @ app.component.ts:8
__webpack_require__ @ bootstrap:81
./src/main.ts @ environment.ts:10
__webpack_require__ @ bootstrap:81
0 @ main.ts:11
__webpack_require__ @ bootstrap:81
checkDeferredModules @ bootstrap:43
webpackJsonpCallback @ bootstrap:30
(anonymous) @ main.js:1

This is what my dependencies looks like:

```json
{
  "dependencies": {
    "@angular/animations": "^6.0.0",
    "@angular/common": "^6.0.0",
    "@angular/compiler": "^6.0.0",
    "@angular/core": "^6.0.0",
    "@angular/forms": "^6.0.0",
    "@angular/http": "^6.0.0",
    "@angular/platform-browser": "^6.0.0",
    "@angular/platform-browser-dynamic": "^6.0.0",
    "@angular/router": "^6.0.0",
    "angular2-multiselect-dropdown": "^2.5.0",
    "core-js": "^2.5.4",
    "font-awesome": "^4.7.0",
    "it-mlaf-sass": "^5.0.0",
    "mydatepicker": "^2.6.1",
    "ng2-dual-list-box": "^1.0.5",
    "ngx-bootstrap": "^2.0.2",
    "ngx-infinite-scroll": "^0.8.3",
    "ngx-loading": "^1.0.8",
    "rxjs": "^6.1.0",
    "rxjs-compat": "^6.1.0",
    "xlsx": "^0.12.9",
    "zone.js": "^0.8.26"
  },
```

**Comment by SheetJSDev** [comment]
This was fixed in version 0.12.12, see https://github.com/SheetJS/js-xlsx/issues/1088

**Comment by INGCRENGIFO** [comment]
npm install xlsx@0.12.12
